src/System.Private.CoreLib/src/System/Reflection/Emit/TypeBuilder.cs

```diff
@@ -605,10 +584,12 @@ private FieldBuilder DefineDataHelper(string name, byte[] data, int size, FieldA
             strValueClassName = ModuleBuilderData.MultiByteValueClass + size.ToString();
 
             // Is this already defined in this module?
-            Type temp = m_module.FindTypeBuilderWithName(strValueClassName, false);
-            valueClassType = temp as TypeBuilder;
-
-            if (valueClassType == null)
+            Type? temp = m_module.FindTypeBuilderWithName(strValueClassName, false);
+            if (temp is TypeBuilder)
+            {
+                valueClassType = (TypeBuilder)temp;
+            }
+            else
```

Comment:
Why change this?

Reply:
compiler was complaining on `valueClassType = temp as TypeBuilder`. Will update this variable to be nullable and keep old code